### PR TITLE
Stops from catbanned folks getting that sweet golem rune.

### PR DESCRIPTION
**Pull request by ghost**
Does what is says on the tin.

### CODE REVIEW

**Comment by TheOneAndOnlyCreeperJoe** [comment]
Code's good.

**Comment by ghost** [comment]
I don't know if this will runtime if a braindead catbanned ghost is on a rune. I'm gonna change the check to the next check when there is a client.

File: code/modules/research/xenobiology/xenobiology.dm [hunk 1]
```diff
@@ -403,7 +403,7 @@
 /obj/effect/golemrune/attack_hand(mob/living/user)
 	var/mob/dead/observer/ghost
 	for(var/mob/dead/observer/O in src.loc)
-		if(!O.client)	continue
+		if(!O.client && !jobban_isbanned(O, "catban"))	continue
```
Comment (JohnGinnane) [inline code/modules/research/xenobiology/xenobiology.dm]: Two things:
1. This means that if the observer isn't a client AND they are not catbanned then it won't let them play as a golem. I take it this should be "OR they're catbanned"
2. Should we also exclude cluwnebanned people too?

**Comment by ghost** [comment]
Cluwnebans probably should be added.
Shouldn't you be checking there is nothing that should make them fail. If one or both trues are false then it doesn't continue right?
 if(!O.client && !jobban_isbanned(O, "catban")) continue

**Comment by ghost** [comment]
~~Did cluwnebans actually ever get fixed?~~
Apparently so, but I'm dumb and didn't know.

**Comment by ghost** [comment]
Hold up I broke something.

**Comment by ghost** [comment]
For some reason, the check only works in the first if statement?

**Comment by ghost** [comment]
It was working then it broke somehow. This is sub-optimal.

**Comment by trifornt-tinfornt** [comment]
Regardless, this is the best thing ever. You have no idea how obnoxious it is for some catbeast to show up as a Golem, run around spamming scream and promptly getting lynched by the other golems which is not only making me lose a golem but is annoying the hell out of me and wasting my time.

**Comment by dreamcrab** [comment]
when this isn't broken, this is a very good idea

**Comment by ghost** [comment]
> yfw CONTINUE doesn't act as continue on IF statement logic but as a CONTINUE TO THE NEXT ELEMENT of the LOOP

I think after a proper code review this should be good to merge after one more HA approval.

**Comment by Nebyab** [comment]
someone code review this.

otherwise its fine

File: code/modules/research/xenobiology/xenobiology.dm [hunk 2]
```diff
@@ -393,6 +393,7 @@
 	for(var/mob/dead/observer/O in src.loc)
 		if(!O.client)	continue
 		if(O.mind && O.mind.current && O.mind.current.stat != DEAD)	continue
+		if(!(jobban_isbanned(O, "catban") || jobban_isbanned(O, "cluwneban")))	continue
```
Comment (Kokojos) [inline code/modules/research/xenobiology/xenobiology.dm]: || is ''or'', so you're basicly doing :
If person has been EITHER not catbanned or not cluwnebanned, continue.

It should be, if either is true, then stop the thing.

if(jobban_isbanned(O, "catban")) ||  jobban_isbanned(O, "cluwneban") return

If jobanned from either, return null.

Comment (JohnGinnane) [inline code/modules/research/xenobiology/xenobiology.dm]: Remove the ! so you have the following:

`if (jobban_isbanned(O, "catban") || jobban_isbanned(I, "cluwneban")) continue`

This means if they're catbanned OR they're cluwnebanned then jump straight to the next iteration of the loop (and  do not execute anything after)

File: code/modules/research/xenobiology/xenobiology.dm [hunk 3]
```diff
@@ -404,7 +405,8 @@
 	var/mob/dead/observer/ghost
 	for(var/mob/dead/observer/O in src.loc)
 		if(!O.client)	continue
-		if(O.mind && O.mind.current && O.mind.current.stat != DEAD)	continue
+		if(O.mind && O.mind.current  && O.mind.current.stat != DEAD)	continue
+		if(!(jobban_isbanned(O, "catban") || jobban_isbanned(O, "cluwneban")))	continue
```
Comment (JohnGinnane) [inline code/modules/research/xenobiology/xenobiology.dm]: See comment above. Same scenario here

**Comment by ghost** [comment]
Literally about to do that, I think I had it working and forgot to commit it.

**Comment by ghost** [comment]
Woowee, I fixed it.

**Comment by Kokojos** [comment]
finnaly

**Pull request merged**